htdocs/variants/class/ProductCombination.class.php

```diff
@@ -226,20 +233,16 @@ public function saveCombinationPriceLevels($clean = 1)
 		}
 	}
 
-
 	/**
-	 * Retrieves information of a variant product and ID of its parent product.
+	 * Return the product id of the parent product of a variant product (Get fk_product_parent by fk_product_child)
 	 *
-	 * @param 	int 	$productid 				Product ID of variant
-	 * @param	int		$donotloadpricelevel	Avoid loading price impact for each level. If PRODUIT_MULTIPRICES is not set, this has no effect.
-	 * @return 	int 							<0 if KO, 0 if product ID is not ID of a variant product (so parent not found), >0 if OK (ID of parent)
+	 * @param int $fk_child 	Product row id
+	 * @return int 				>0 if OK, 0 if product is not a variant, <0 if KO
 	 */
-	public function fetchByFkProductChild($productid, $donotloadpricelevel = 0)
+	public function getFkProductParentByFkProductChild($fk_child)
```

Comment:
Instead of replacing the function fetchByFkProductChild() by getFkProductParentByFkProductChild() that do something else
why don't just just call the function fetchByFkProductChild() by adding the second parameter to 1. It seems it does the same thing than the new getFkProductParentByFkProductChild()

Reply:
In this modification, I position myself as a data synchronizer between Dolibarr and one or more online stores.

Actually, there are 2 choices:

My thought was: because it is possible to have multiples combinations between a parent and a child (by forcing the reference when creating combination), then fetchByFkProductChild is deprecated. That's why I also created the fetchAllByFkProductChild method which returns all the combinations related to a child product. I initially thought it was a good idea. Several stores can synchronize their own attributes, so it makes sense that they can create their own combinations of products, using the same references. Looking back, I wonder if it's really useful.

The second way is: to force to have only one combination between a parent and a child. So, if several stores use the same references, they no longer import their own attributes, but use the existing reference as is. (ideally they use the same attributes)

Reply:
@eldy What do you think? I can modify the code to add only the ref_ext. Which will be of little use in this case because the interest, ultimately, was to allow each online store to identify its own combinations. Or, abandon this PR, and modify the PR #14633 to keep the improvements useful, without ref_ext.

Reply:
I still don't see any difference in code between you new function getFkProductParentByFkProductChild and the old one fetchByFkProductChild when second parameter is 1.
Signature entry is same, code content is similar too and return code too. So i still don't understand why you removed it and introduce another one that do exactly the same ? Can you keep old name fetchByFkProductChild instead of renaming it into getFkProductParentByFkProductChild ?